test/unit/Signer.test.js

```diff
@@ -82,10 +111,17 @@ describe('Signer', () => {
                     field: 'some-data',
                 },
                 1,
-                '0xF915eD664e43C50eB7b9Ca7CfEB992703eDe55c4',
+                address,
                 '0x3d5c221ebed6bf75ecd0ca8751aa18401ac60561034e3b2889dfd7bbc0a2ff3c5f1c5239113f3fac5b648ab665d152ecece1daaafdd3d94309c2b822ec28369e1c',
             )
-            assert.throws(() => Signer.verifyStreamMessage(streamMessage), Error)
+            assert.throws(() => Signer.verifyStreamMessage(streamMessage, new Set([address])), Error)
+        })
+
+        it('Should throw if correct signature but not from a trusted publisher', async () => {
+            const streamMessage = new StreamMessage()
+            Signer.getPayloadToSign = sinon.stub().returns('')
```

Comment:
Whoopsie! You're overwriting the static method on the class. The class will remain crippled and any test case that comes after this one will fail spectacularly.

It might not be a great idea anyway to stub internals of the class under test: better would be the usual pattern of providing valid input to the public method that you're testing (like you're doing in the other tests).

To avoid repeating the code to create a signed message to test, just create one in the `beforeEach` block and use that in the test cases, tweaking it to suit the particular test case in question (mess up the signature etc.).